lib/fizzy/parser.cpp

```diff
@@ -81,6 +81,47 @@ struct parser<Global>
     }
 };
 
+template <>
+struct parser<Import>
+{
+    parser_result<Import> operator()(const uint8_t* pos)
+    {
+        std::vector<uint8_t> module;
+        std::tie(module, pos) = parser<std::vector<uint8_t>>{}(pos);
+        std::vector<uint8_t> name;
+        std::tie(name, pos) = parser<std::vector<uint8_t>>{}(pos);
+
+        Import result{};
+        result.module.assign(module.begin(), module.end());
+        result.name.assign(name.begin(), name.end());
+
+        const uint8_t type = *pos++;
+        switch (type)
+        {
+        case 0x00:
+            result.type = ImportType::Function;
+            std::tie(result.desc.function_type_index, pos) = leb128u_decode<uint32_t>(pos);
+            break;
+        case 0x01:
+            result.type = ImportType::Table;
+            throw parser_error{"importing Tables is not implemented"};
+            break;
+        case 0x02:
+            result.type = ImportType::Memory;
+            std::tie(result.desc.memory, pos) = parser<Memory>{}(pos);
+            break;
+        case 0x03:
+            result.type = ImportType::Global;
+            std::tie(result.desc.global, pos) = parser<Global>{}(pos);
```

Comment:
I realized this is not correct, as imported globals don't have initializer expression, so we can't reuse `Global` parser here. I'll change this.

Reply:
Fixed by overwriting commits.